test/Mono.Linker.Tests.Cases/Reflection/ExpressionNewType.cs

```diff
@@ -20,8 +42,38 @@ class T1
 			[Kept]
 			internal T1 ()
 			{
-
 			}
 		}
+
+		class T2
+		{
+		}
+
+		[Kept]
+		static void Branch_SystemTypeValueNode ()
+		{
+			var expr = Expression.New (typeof (T1));
+		}
+
+		[Kept]
+		static void Branch_NullValueNode ()
+		{
+			var expr = Expression.New (Helper1 () + Helper2 () == 12 ? null : typeof (T1));
```

Comment:
This test sort of breaks the test above. The tests rely on the `T1` being referenced only from one place - and the validation then checks that linker found the dependency.
So by reusing the type here, it's hard to tell if all the cases work, since if only one of them does, the others won't be validated.

So you need to create a new type for each test case.

Same logic applies to methods/fields/properties in the other tests.